server.cpp

```diff
@@ -107,48 +123,32 @@ void Server::run(){
       std::cout << "--------ERROR-------Boost Error Code-----" << error << std::endl;
       return;
     }
-    std::string request(req_buf);
+    std::string raw_request(req_buf);
     std::string errorMessage;
     
-    HttpRequest httpReq(request);
-    HttpResponse* response;
-    if (!httpReq.parse()){
-      return; 
-    }
-
-    std::string requested_path = httpReq.getPath();
-    auto it = path_to_handler.find(requested_path);
-    if (it == path_to_handler.end())
-      _handler = new RequestHandler();
-     
-    else if (it->second.first == "EchoHandler")
-      _handler = new EchoHandler();
-
-    else if (it->second.first == "StaticHandler")
-      _handler = new StaticHandler(it->second.second);
-
-    else
-      _handler = new RequestHandler();
-
-    bool handle_status;
-    handle_status = _handler->handle_request(httpReq, response);
+    auto httpRequest = Request::Parse(raw_request);
+    Response  * httpResponse  = new Response();;
 
-    if (!handle_status || response == NULL){
-      std::cout << "now down here" << std::endl;
-      if (!response)
-        std::cout << "response is null " << std::endl;
-      break;
-   }
-    std::string response_str = response->toString();
-    std::size_t bytes_written = socket.write_some(boost::asio::buffer(response_str), error);
+    //TODO: This extraction doesn't work with a StaticHandler path of "/"
+    std::string uri = httpRequest->uri();
+    std::size_t prefix_slash = uri.find("/", 1);
+    std::string uri_prefix = uri.substr(0, prefix_slash);
+  
+    RequestHandler::Status response_status;
+    response_status = _handlerContainer[uri_prefix]->HandleRequest((*httpRequest), httpResponse);
```

Comment:
Line 138 needs to be fixed. We should be calling _handlerContainer.find(uri_prefix) first to ensure a handler is mapped to that prefix first. If none is found we should invoke the default handler as specified in the config